server/exposures/cmaq.py

```diff
@@ -122,33 +98,68 @@ def get_scores(self, **kwargs):
+        utc_time = pytz.timezone('UTC').localize(datetime.strptime(kwargs.get('startDate'), "%Y-%m-%d"))
+        start_time = tzone.localize(datetime.strptime(kwargs.get('startDate'), "%Y-%m-%d"))
+        end_time = tzone.localize(datetime.strptime(kwargs.get('endDate'), "%Y-%m-%d"))
+        time_diff = utc_time - start_time
+        # retrieve query result for each lat,lon pair and add to data object
         lat_lon_set = kwargs.get('latLon').split(';')
         for lat_lon in lat_lon_set:
-            exposure_time = tzone.localize(datetime.strptime(kwargs.get('startDate'), "%Y-%m-%d"))
-            end_time = tzone.localize(datetime.strptime(kwargs.get('endDate'), "%Y-%m-%d"))
-
-            while exposure_time <= (end_time + timedelta(hours=23)):
-                data['scores'] += [{'dateTime': exposure_time,
+            coords = lat_lon.split(',')
+            row, col = latlon2rowcol(coords[0], coords[1], str(start_time.year))
+            session = Session()
+            if kwargs.get('resolution') == 'hour':
+                # hourly resolution of data - return all hours for date range
+                query = session.query(CmaqExposuresDatum.id,
+                                      CmaqExposuresDatum.utc_date_time,
+                                      getattr(CmaqExposuresDatum, exposure)). \
+                    filter(CmaqExposuresDatum.utc_date_time >= start_time). \
+                    filter(CmaqExposuresDatum.utc_date_time <= end_time + timedelta(hours=23)). \
+                    filter(CmaqExposuresDatum.row == row). \
+                    filter(CmaqExposuresDatum.col == col)
+            else:
+                # daily resolution of data - return only matched hours for date range
+                query = session.query(CmaqExposuresDatum.id,
+                                      CmaqExposuresDatum.utc_date_time,
+                                      getattr(CmaqExposuresDatum, exposure)). \
+                    filter(CmaqExposuresDatum.utc_date_time >= start_time). \
+                    filter(CmaqExposuresDatum.utc_date_time <= end_time). \
+                    filter(CmaqExposuresDatum.row == row). \
+                    filter(CmaqExposuresDatum.col == col). \
+                    filter(extract('hour', CmaqExposuresDatum.utc_date_time) ==
+                           ((start_time - utc_time).seconds // 3600))
+            session.close()
+            # add query output to data object in JSON structured format
+            for id, adj_date_time, exp in query:
+                # print(id, adj_date_time, exp)
+                data['scores'] += [{'dateTime': tzone.localize(adj_date_time + time_diff),
                                     'latLon': lat_lon,
-                                    'score': str(round(uniform(1,5),4))}]
-                exposure_time += timedelta(hours=time_delta)
-
+                                    'score': str(cmaq_calc_score(kwargs.get('exposureType'),exp))}]
```

Comment:
data['scores'] is a list, so perhaps ```append``` should be used here rather than ```+```? Like data['scores'].append({'dateTime':..., ...})?

Reply:
This makes sense, will adjust